src/django/api/views.py

```diff
@@ -159,6 +164,65 @@ def all_countries(request):
     return Response(COUNTRY_CHOICES)
 
 
+class FacilitiesViewSet(ReadOnlyModelViewSet):
+    queryset = Facility.objects.all()
+    serializer_class = FacilitySerializer
+    permission_classes = (AllowAny,)
+
+    def list(self, request):
+        name = request.query_params.get(FacilitiesQueryParams.NAME,
+                                        None)
+        contributors = request.query_params \
+                              .getlist(FacilitiesQueryParams.CONTRIBUTORS)
+        contributor_types = request \
+            .query_params \
+            .getlist(FacilitiesQueryParams.CONTRIBUTOR_TYPES)
+        countries = request.query_params \
+                           .getlist(FacilitiesQueryParams.COUNTRIES)
+
+        queryset = Facility.objects.all()
+
+        if name is not None:
+            queryset = queryset.filter(name__icontains=name)
+
+        if countries is not None and len(countries):
+            queryset = queryset.filter(country_code__in=countries)
+
+        if len(contributor_types):
+            type_match_facility_ids = [
+                match.facility.id
```

Comment:
We could probably use a [values QuerySet](https://docs.djangoproject.com/en/2.0/ref/models/querysets/#django.db.models.query.QuerySet.values) here rather to avoid hydrating a bunch of model instances when we only want the ID.

```python
 [int(f['facility__id']) for f in FacilityMatch.objects.filter(...).values('facility__id')]
```

Reply:
Sounds good! I made this change in 9c62667e2f896af9a63df28a7636edb7ca7de292